### PR TITLE
Add support for multiple --ignore-path options

### PR DESCRIPTION
**Pull request by tgriesser**
Thought this might be useful, tried to use both `.gitignore` and `.prettierignore` and saw it didn't work.

### CODE REVIEW

File: bin/prettier.js [hunk 1]
```diff
@@ -454,15 +454,18 @@ function writeOutput(result, options) {
 function eachFilename(patterns, callback) {
   // The ignorer will be used to filter file paths after the glob is checked,
   // before any files are actually read
-  const ignoreFilePath = path.resolve(ignorePath);
-  let ignoreText = "";
+  const ignorePaths = Array.isArray(ignorePath) ? ignorePath : [ignorePath];
```
Comment (azz) [inline bin/prettier.js]: I usually use `[].concat(ignorePath)`

**Comment by azz** [comment]
Is there precedent for this in other projects?

**Comment by tgriesser** [comment]
> Is there precedent for this in other projects?

I'd say it's common to supply options more than once and result in all of those options being used, particularly when the option is something like an include/exclude.

eslint: [--ext](https://eslint.org/docs/user-guide/command-line-interface#--ext), [--rulesdir](https://eslint.org/docs/user-guide/command-line-interface#--rulesdir), [--ignore-pattern](https://eslint.org/docs/user-guide/command-line-interface#--ignore-pattern)
nodemon: [--ignore](https://github.com/remy/nodemon#ignoring-files)

File: bin/prettier.js (hunk 1 above)
Comment (tgriesser) [reply bin/prettier.js]: @azz Updated.

File: bin/prettier.js [hunk 2]
```diff
@@ -454,15 +454,18 @@ function writeOutput(result, options) {
...
-    ignoreText = fs.readFileSync(ignoreFilePath, "utf8");
-  } catch (readError) {
-    if (readError.code !== "ENOENT") {
-      console.error(`Unable to read ${ignoreFilePath}:`, readError);
-      process.exit(2);
+  let ignoreText = "";
+  for (let i = 0; i < ignorePaths.length; i++) {
+    const ignoreFilePath = path.resolve(ignorePaths[i]);
+    try {
+      ignoreText += "\n" + fs.readFileSync(ignoreFilePath, "utf8");
```
Comment (azz) [inline bin/prettier.js]: It looks like `ignore` supports an array of ignore patterns. https://www.npmjs.com/package/ignore#addpatterns

```js
const ignoreTexts = ignorePaths.map(path.resolve).map(ignoreFilePath => {
   ... existing file read code ...
});

const ignore = ignore().add(ignoreTexts);
```

**Comment by azz** [comment]
Just FYI, once #2730 is merged there will be conflicts here.

**Comment by lydell** [comment]
@tgriesser I think this can be a useful feature. Would you like to update this?

**Comment by azz** [comment]
Closing due to inactivity. Please re-open if you're able to address the review comments and resolve conflicts.

**Comment by bennyn** [comment]
By default `.prettierignore` is parsed to skip individual files and directories. But is it true that a definiton of `--ignore-path` overwrites this behaviour?

In other words... Will the `.prettierignore` file be skipped when defining something like `prettier --ignore-path .gitignore`?

**Pull request closed without merging**